src/MongoDB.EntityFrameworkCore/Serializers/EntitySerializer.cs

```diff
@@ -48,7 +53,49 @@ public TValue Deserialize(BsonDeserializationContext context, BsonDeserializatio
             => Deserialize(context, args);
 
         public void Serialize(BsonSerializationContext context, BsonSerializationArgs args, TValue value)
-            => throw new NotImplementedException();
+        {
+            if (value == null)
+            {
+                context.Writer.WriteNull();
+                return;
+            }
+
+            context.Writer.WriteStartDocument();
+            WriteKey(context.Writer, value);
+            WriteProperties(context.Writer, value, _entityType.GetProperties().Where(p => !p.IsPrimaryKey() && _isStored(p)));
+            context.Writer.WriteEndDocument();
+        }
+
+        private void WriteKey(IBsonWriter writer, TValue entry)
+        {
+            var keyProperties = _entityType.FindPrimaryKey()?.Properties.Where(_isStored).ToArray() ?? [];
+
+            var hasCompoundKey = keyProperties.Length > 1;
+            if (hasCompoundKey)
+            {
+                writer.WriteName("_id");
+                writer.WriteStartDocument();
+            }
+
+            WriteProperties(writer, entry, keyProperties);
+
+            if (hasCompoundKey)
+            {
+                writer.WriteEndDocument();
+            }
+        }
+
+        private static void WriteProperties(IBsonWriter writer, TValue entry, IEnumerable<IReadOnlyProperty> properties)
+        {
+            foreach (var property in properties)
+            {
+                var serializationInfo = SerializationHelper.GetPropertySerializationInfo(property);
+                var elementName = serializationInfo.ElementPath?.Last() ?? serializationInfo.ElementName;
+                writer.WriteName(elementName);
+                var root = BsonSerializationContext.CreateRoot(writer);
```

Comment:
We usually call this variable context but you're free to keep whatever name you want.

Reply:
Will rename.